homeassistant/components/lg_thinq/entity.py

```diff
@@ -29,43 +33,42 @@ def __init__(
         self,
         coordinator: DeviceDataUpdateCoordinator,
         entity_description: EntityDescription,
-        property: ThinQProperty,
+        idx: str,
     ) -> None:
         """Initialize an entity."""
         super().__init__(coordinator)
 
         self.entity_description = entity_description
-        self.property = property
+        self.idx = idx
+
         self._attr_device_info = dr.DeviceInfo(
             identifiers={(DOMAIN, coordinator.unique_id)},
             manufacturer=COMPANY,
-            model=coordinator.device_api.model_name,
+            model=coordinator.api.device.model_name,
             name=coordinator.device_name,
         )
-
-        # Set the unique key. If there exist a location, add the prefix location name.
-        unique_key = (
-            f"{entity_description.key}"
-            if property.location is None
-            else f"{property.location}_{entity_description.key}"
-        )
-        self._attr_unique_id = f"{coordinator.unique_id}_{unique_key}"
+        self._attr_unique_id = f"{coordinator.unique_id}_{self.idx}"
```

Comment:
Does the unique_id change now? Is it still stable? What is idx?
```suggestion
        self._attr_unique_id = f"{coordinator.unique_id}_{idx}"
```

Reply:
Actually we referred to the Home Assistant Developer page.
- https://developers.home-assistant.io/docs/integration_fetching_data/

Reply:
so what is idx then, just a number of order?

Reply:
Yes it is.

Reply:
But the unique_id should _never_ change. Otherwise it will create new entities and break peoples automations. So if I understand, if I delete a device, then the device with `idx` 2, could turn into `idx` 1 after a next reboot. And then the device would be shown as a complete new device, we should avoid that at all costs!

So ideally unique_id consists of something unique (which we have iirc), and something to identify the entity (mostly in the case of binary sensors or sensors, where there can be more entities per device). A unique_id is unique per platform per integration, so you don't have to add prefixes for the platform to it.

Reply:
Let me check if the idx changes after deletion and adding.

Reply:
Let me clarify that.
'idx' is generated in the thinqconnect library based on the property key.
So it doesn't change after device removal and addition again.
And 'unique_id' of the coordinator is unique and static because it is generated based on device_id and sub_id.
In conclusion, _attr_unique_id is unique and static. So it doesn't cause any trouble in automations.

Reply:
Okay awesome